samza-core/src/main/java/org/apache/samza/execution/JobGraph.java

```diff
@@ -183,11 +184,11 @@ StreamEdge getOrCreateStreamEdge(StreamSpec streamSpec) {
    * @param isIntermediate  boolean flag indicating whether it's an intermediate stream
    * @return stream edge
    */
-  StreamEdge getOrCreateStreamEdge(StreamSpec streamSpec, boolean isIntermediate) {
+  private StreamEdge getOrCreateStreamEdge(StreamSpec streamSpec, boolean isIntermediate) {
     String streamId = streamSpec.getId();
     StreamEdge edge = edges.get(streamId);
     if (edge == null) {
-      boolean isBroadcast = specGraph.getBroadcastStreams().contains(streamId);
```

Comment:
@cameronlee314 I'll address this in the broadcast stream PR. There are two sources of broadcast inputs: user defined inputs and those added by the high level API's broadcast method. This is tracking the latter.

Reply:
I actually deleted it for now. @prateekm you can add it later if proper.

Reply:
@nickpan47 Deleting this will break existing support for high level API broadcast streams. If we keep this, adding support for declaring input descriptors is an additive step that we can choose to do after the next release. If we remove this, it'll be a must have. So would prefer to keep this for now.

Reply:
Discussed with @prateekm and will leave it as it now for later complete fix of the broadcast stream configuration.